bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusVMB8INHandler.java

```diff
@@ -0,0 +1,293 @@
+
+        refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+            sendCounterStatusRequest(velbusBridgeHandler, ALL_CHANNELS);
+        }, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+
+        VelbusBridgeHandler velbusBridgeHandler = getVelbusBridgeHandler();
+        if (velbusBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            if (channelUID.equals(counter1Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x01);
+            } else if (channelUID.equals(counter2Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x02);
+            } else if (channelUID.equals(counter3Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x04);
+            } else if (channelUID.equals(counter4Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x08);
+            } else if (channelUID.equals(counter5Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x10);
+            } else if (channelUID.equals(counter6Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x20);
+            } else if (channelUID.equals(counter7Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x40);
+            } else if (channelUID.equals(counter8Channel)) {
+                sendCounterStatusRequest(velbusBridgeHandler, (byte) 0x80);
+            }
+        }
+    }
+
+    protected void sendCounterStatusRequest(VelbusBridgeHandler velbusBridgeHandler, byte channel) {
+        VelbusCounterStatusRequestPacket packet = new VelbusCounterStatusRequestPacket(
+                new VelbusChannelIdentifier(getModuleAddress().getAddress(), channel));
+
+        byte[] packetBytes = packet.getBytes();
+        velbusBridgeHandler.sendPacket(packetBytes);
+    }
+
+    @Override
+    public boolean onPacketReceived(byte[] packet) {
+        if (!super.onPacketReceived(packet)) {
+            return false;
+        }
+
+        if (packet[0] == VelbusPacket.STX && packet.length >= 5) {
+            byte command = packet[4];
+
+            if (command == COMMAND_COUNTER_VALUE && packet.length >= 12) {
+                int counterChannel = (packet[5] >> 4) & 0x07;
+
+                double counterValue = ((double) (((packet[8] & 0xff) << 24) | ((packet[9] & 0xff) << 16)
+                        | ((packet[10] & 0xff) << 8) | (packet[11] & 0xff)));
+                double currentValue = ((double) (((packet[6] & 0xff) << 8) | (packet[7] & 0xff)));
+
+                switch (counterChannel) {
+                    case 0x00:
+                        if ("kWh".equals(vmb8inConfig.counter1Unit)) {
```

Comment:
Could you extract these units (magix strings0  to class level string constants? (class wide request)

Reply:
Done in last update.